jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/content/settlers/profession/ProfessionPanel.java

```diff
@@ -1,66 +1,245 @@
+import jsettlers.common.map.partition.IProfessionSettings;
+import jsettlers.common.movable.EMovableType;
+import jsettlers.common.player.IInGamePlayer;
 import jsettlers.common.position.ShortPoint2D;
+import jsettlers.graphics.action.ActionFireable;
+import jsettlers.graphics.localization.Labels;
 import jsettlers.graphics.map.controls.original.panel.content.AbstractContentProvider;
-import jsettlers.graphics.map.controls.original.panel.content.BarFill;
 import jsettlers.graphics.map.controls.original.panel.content.ESecondaryTabType;
 import jsettlers.graphics.map.controls.original.panel.content.updaters.UiContentUpdater;
 import jsettlers.graphics.map.controls.original.panel.content.updaters.UiLocationDependingContentUpdater;
+import jsettlers.graphics.ui.Button;
 import jsettlers.graphics.ui.Label;
+import jsettlers.graphics.ui.Label.EHorizontalAlignment;
+import jsettlers.graphics.ui.UIElement;
 import jsettlers.graphics.ui.UIPanel;
-import jsettlers.graphics.ui.layout.ProfessionsLayout;
 
 public class ProfessionPanel extends AbstractContentProvider implements UiContentUpdater.IUiContentReceiver<IPartitionSettings> {
 
-	private UIPanel panel;
-	private final UiLocationDependingContentUpdater<IPartitionSettings> uiContentUpdater = new UiLocationDependingContentUpdater<>(this::currentDistributionSettingsProvider);
+	private final ContentPanel panel;
+	private final UiLocationDependingContentUpdater<IPartitionSettings> uiContentUpdater;
+	private long lastChangeTimestamp;
 
 	public ProfessionPanel() {
-		panel = new ProfessionsLayout()._root;
-		uiContentUpdater.addListener(this);
+		this.panel = new ContentPanel();
+		this.uiContentUpdater = new UiLocationDependingContentUpdater<>(ProfessionPanel::currentDistributionSettingsProvider);
+		this.uiContentUpdater.addListener(this);
+		this.lastChangeTimestamp = 0;
 	}
 
 	@Override
 	public UIPanel getPanel() {
 		return panel;
 	}
 
-	private IPartitionSettings currentDistributionSettingsProvider(IGraphicsGrid grid, ShortPoint2D position) {
-		if (grid.getPlayerAt(position.x, position.y) != null) {
-			return grid.getPartitionData(position.x, position.y).getPartitionSettings();
-		}
+	@Override
+	public ESecondaryTabType getTabs() {
+		return ESecondaryTabType.SETTLERS;
+	}
 
-		return null;
+	@Override
+	public void showMapPosition(ShortPoint2D position, IGraphicsGrid grid) {
+		this.panel.setPosition(position);
+		super.showMapPosition(position, grid);
+		this.uiContentUpdater.updatePosition(grid, position);
 	}
 
 	@Override
-	public ESecondaryTabType getTabs() {
-		return ESecondaryTabType.SETTLERS;
+	public void contentShowing(ActionFireable actionFireable) {
+		super.contentShowing(actionFireable);
+		this.uiContentUpdater.start();
 	}
 
 	@Override
-	public void update(IPartitionSettings data) {
+	public void update(IPartitionSettings partitionSettings) {
+		// Use delay to prevent race condition between direct ui update and settings update
```

Comment:
This is an ugly workaround.
Please  refactor SetMoveableRatioAction to not transfer the new value but just the delta.

Reply:
I am not sure if this would fix the problem. There is a delay because the change of the ratio is scheduled as a task and the updater of the ui is also working by a timer. I am updating the ui directly so that the user has a fast feedback for his click and after the click this workaround prevents from getting an old update to early (as example if the user clicks multiple times very fast).

Reply:
Are you sure that the ui values are updated on a timer and not on every frame?
I think the issue behind the delay is that there is some latency between tasks being scheduled and actually executed (for multiplayer purposes). You can see this when you give move orders or change the stock settings

Reply:
or even more noticeably in the production settings if you adjust the Barfill for different tools

Reply:
The ui is updated by UiContentUpdater (and there is a delay of 1000). I am not sure if this is the best available way but using it was very confusing. It is very hard to follow the source of the update calls.

Reply:
I looked into it and the delay in UiContentUpdater seams to make a big difference.
This part of the code is not shared with Android so it wouldn't be an issue to lower the delay to maybe 50-100.

Reply:
The more i am looking at UiContentUpdater the more i think this thing should be removed completely and instead a update method should be added to AbstractContentProvider. Then the main panel could simply manage the timer and call the update method. The UiContentUpdate seems to provide very much unnecessary complexity.

Reply:
There are still two types of UiContentUpdater: player and location based. Having an update method per interested ACP sounds like a lot of copy&paste code.

Reply:
At the moment we have the following steps in the content providers for each updater that is used:
- Instantiate the updater and pass the provider
- Register the listener
- Start the updater
- Update the position of the updater
- Update the player of the updater
- Handle the updates
- Stop the updater
Most of this steps could be done in the MainPanel once. This would also prevent from leaking resources like i did (forgot to call the stop method). There will be no copy paste code. The abstract content provider could provide two methods (onPositionUpdate, onPlayerUpdate) that could be overriden if needed.

Reply:
I think this should be the matter of another pr because this aspect of your pr is working as intended and not really related to bearer limit or similar.
I think AbstractContentProvider is meant as an general purpose Content holder and onPositionUpdate and onPlayerUpdate sounds a bit too specific for that.
I would favor some kind of Subclass e.g. PositionDependentContentProvider that has the onPositionUpdate method.
Whats your opinion on that?

Reply:
Yes i totally agree with the separate pull request for this topic.

Additional subclasses like PositionDependentContentProvider is not a good idea because bigger hierarchies of inheritance are a code smell. 
I understand your objection with the general purpose ... but if i see it correctly then the MainPanel is currently the only usecase for the AbstractContentProvider. Rename it to something like MainPanelContentProvider would fix this problem. And i think it is ok to see position and player updates as a general feature of MainPanelContentProvider implementations.
As an alternative it would also be possible to provide interfaces that can be implemented by AbstractContentProvider implementations on demand and then be checked if they are implemented by the MainPanel.
Another alternative would be a update method with a event like parameter. Then different events (PositionUpdatedEvent or PlayerUpdatedEvent) could be passed.

Reply:
regarding the ACP refactoring:
I might be wrong but all classes that request the position actually just want the partition, don't they?
In that case it would be an onUpdatedPartition() function.
The thing that is most important to me is that there is just one place in the code where the partition is checked against the current playerId. This should prevent other players from changing or seeing your settings and statistics without any potential loopholes or duplicate code that needs bug fixes in different places.

AbstractContentProvider -> PartitionDependentContentProvider -> WhateverPanel

This hierarchy isn't very deep for my taste and every class has its own distinct purpose.

but about the main topic:
I think my problem with your timer+absolute value is that
the arrows have their use case in increasing or decreasing some value
but only the value holder (in this case the partition) itself decides what the current value actually is and therefore what plus or minus five percent of the current value is as an absolute value.
The ProfessionsPanel does not actually know what the value will be in the future as its not related to the partition.

If you press the up arrow one or two times the ProfessionsPanel will show `x x+5 (x+10)`
but the partition might not apply the changes in five seconds because the game is paused or just very unresponsive.
In this case the ProfessionsPanel will just show the old value `x` again.
If you now press the up arrow and unpause the game afterwards the ProfessionsPanel will show and apply `x+5`.
This wouldn't make sense for my as a player if I press the same button three times but it changes just as if I pressed only one time.

Reply:
> AbstractContentProvider -> PartitionDependentContentProvider -> WhateverPanel

The problem with this hierarchy is that you add a hierarchy step to add a feature. If you would add more features this way then your hierarchy would end up bigger and bigger. It is not the right answer to add features. A better tool to add features is composition or subtyping (not to be confused with implementation inheritance). You can read more about this here:
- https://www.yegor256.com/2016/09/13/inheritance-is-procedural.html
- https://www.infoworld.com/article/2073649/why-extends-is-evil.html

> If you press the up arrow one or two times the ProfessionsPanel will show x x+5 (x+10)
but the partition might not apply the changes in five seconds because the game is paused or just very unresponsive.
In this case the ProfessionsPanel will just show the old value x again.
If you now press the up arrow and unpause the game afterwards the ProfessionsPanel will show and apply x+5.
This wouldn't make sense for my as a player if I press the same button three times but it changes just as if I pressed only one time.

In my solution the value is directly updated in the uis data model (without any delay) and then later synced with the value of the real model. Only in very rare cases there should be a difference between the uis data model and the value from the real model. In this case the values will be back in sync after a short delay. I think this is acceptable and better then when the user has no feedback or a very late feedback caused by a paused or unresponsive game.

Reply:
My problem is that the ui is not supposed to have a data model. The ui is only about showing and editing settings.
Your system does two things: show the current value or show what value is going to get applied next without any difference between these two indicators.
You system also does not update the value when the camera changes partition after you clicked the arrow buttons.
Your system has only one actual benefit and that would be a little bit more responsiveness if you change the bearer limit, etc. and then do something other than change it on another partition.
This is probably not an issue in most cases but will frequently happen for some players on some maps (with two islands per player).

Yes, the current way its a bit unresponsive and might lead to irritation at first but its overall cleaner in design and doesn't have any buggy edge cases (as far as I am aware).
Also reducing the UPDATE_INTERVAL in UiContentUpdater (as long as it still exists) helps a lot with the responsiveness.

Reply:
I removed the direct ui update hack and reduced the UPDATE_INTERVAL to 100. 
You are right it is better to have a unified design here.